lib/filter_h264.ex

```diff
@@ -0,0 +1,52 @@
+defmodule RecordingConverter.FilterH264 do
+  @moduledoc false
+
+  use Membrane.Filter
+
+  alias Membrane.Buffer
+
+  def_input_pad :input, accepted_format: _accepted_format
+
+  def_output_pad :output, accepted_format: _accepted_format
+
+  @impl true
+  def handle_init(_ctx, _options) do
+    {[], []}
+  end
+
+  @impl true
+  def handle_buffer(_pad, %Buffer{} = buffer, _ctx, state) do
+    size = Enum.count(state)
+
+    cond do
+      buffer.metadata.h264.type in [:sps, :pps] ->
+        {[], [buffer | state]}
+
+      size > 0 ->
+        buffers = Enum.take(state, 2)
+
+        buffers =
+          if size == 2 do
+            Enum.reverse(buffers)
+          else
+            buffers
+          end
+
+        buffers = buffers ++ [buffer]
+
+        {buffers_to_actions(buffers), []}
+
+      true ->
+        {[buffer: {:output, buffer}], state}
+    end
```

Comment:
It would be simpler and more generic if we would keep in state last `:sps` and `:pps` separately.